src/process_utility.c

```diff
@@ -1191,13 +1177,15 @@ process_alter_column_type_end(Hypertable *ht, AlterTableCmd *cmd)
 {
 	ColumnDef  *coldef = (ColumnDef *) cmd->def;
 	Oid			new_type = TypenameGetTypid(typename_get_unqual_name(coldef->typeName));
-	CatalogSecurityContext sec_ctx;
+	Dimension  *dim = hyperspace_get_dimension_by_name(ht->space, DIMENSION_TYPE_ANY, cmd->name);
 
-	catalog_become_owner(catalog_get(), &sec_ctx);
+	if (NULL == dim)
+		return;
+
+	dimension_update_type(dim, new_type);
```

Comment:
Though about it, but I'd rather keep mutations on the dimension object/table separate from any logic that follows as a result of that mutation, if that makes sense? Otherwise, I think we risk overloading the dimension type with lots of logic related to other objects.

Reply:
I would actually rather couple the catalog changes with the actual table changes to prevent the two becoming out of sync. But I'll leave this up to you.